src/sentry/utils/appleconnect/appstore_connect.py

```diff
@@ -0,0 +1,229 @@
+        raise ValueError(
+            "Request body not JSON", full_url, response.status_code, response.text
+        ) from e
+
+
+def _get_next_page(response_json) -> str:
+    """
+    Gets the next page url from a app store connect paged response
+    """
+    return safe.get_path(response_json, "links", "next")
+
+
+def _get_appstore_info_paged_data(
+    session: Session, credentials: AppConnectCredentials, url: str
+) -> Generator[Any, None, None]:
+    """
+    Iterate through all the pages from a paged response and concatenate the `data` part of the response
+
+    App store connect share the general format:
+
+    data:
+      - list of elements
+    links:
+      next: link to the next page
+    ...
+
+    The function iterates through all pages (following the next link) until
+    there is no next page, and returns a generator containing all
+    the data in the arrays from each call
+
+    :return: a generator with the contents of all the arrays from each page (flattened).
+    """
+    while url is not None:
+        response = _get_appstore_info(session, credentials, url)
+        data = response["data"]
+        yield from data
+        url = _get_next_page(response)
+
+
+def get_pre_release_version_info(session: Session, credentials: AppConnectCredentials, app_id: str):
+    """
+    Get all prerelease builds version information for an application
+
+    The release build version information has the following structure:
+    platform: str - the platform for the build (e.g. IOS, MAC_OS ...)
+    short_version: str - the short version build info ( e.g. '1.0.1'), also called "train" in starship documentation
+    id: str - the IID of the version
+    versions: vec - a vector with builds
+        version: str - the version of the build (e.g. '101'), looks like the build number
+        id: str - the IID of the build
+
+    NOTE: the pre release version information is identical to the release version information
+    :return: a list of prerelease builds version information (see above)
+    """
+    url = f"v1/apps/{app_id}/preReleaseVersions"
+    data = _get_appstore_info_paged_data(session, credentials, url)
+    result = []
+    for d in data:
+        versions = []
+        v = {
+            "platform": safe.get_path(d, "attributes", "platform"),
+            "short_version": safe.get_path(d, "attributes", "version"),
+            "id": safe.get_path(d, "id"),
+            "versions": versions,
+        }
+        builds_url = safe.get_path(d, "relationships", "builds", "links", "related")
+        for build in _get_appstore_info_paged_data(session, credentials, builds_url):
+            b = {
+                "version": safe.get_path(build, "attributes", "version"),
+                "id": safe.get_path(build, "id"),
+            }
+            versions.append(b)
+        result.append(v)
+
+    return result
+
+
+def get_release_version_info(session: Session, credentials: AppConnectCredentials, app_id: str):
```

Comment:
A quick scan seems to suggest that this is identical to `get_pre_release_version_info` above other than the URL.  Maybe have a common `_get_version_info` method that both call?  (I may have missed some details, if so please ignore)

Reply:
they are similar but not quite, I am constructing the same structure (to make it easier to deal with it) but it is not extracted from exactly the same response.

short_version:  `attributes.version` vs `attributes.versionString`
there are multiple pre release builds vs one release build

So there are 3 places in the function where the functionality differs, I guess I could make one function and handle the two cases but I'm not sure if it would make the code cleaner.